Cannot find scoped module for the very module I'm trying to write types for

### Discussed in https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/64173

<div type='discussions-op-text'>

<sup>Originally posted by **fivecar** January 31, 2023</sup>
I'm currently trying to create a new types package for @kesha-antonov/react-native-background-downloader. I used `npx dts-gen --dt --name @kesha-antonov/react-native-background-downloader --template module`, and have also modified "paths" in tsconfig, according to the docs, in hopes that the test file can find types correctly:
```json
{
    "paths": {
        "@kesha-antonov/*": ["kesha-antonov__*"]
    }
}
```
However, when I run `npm test @kesha-antonov/react-native-background-downloader`, the test file can't seem import from "@kesha-antonov/react-native-background-downloader":
```Terminal
% npm test @kesha-antonov/react-native-background-downloader

> definitely-typed@0.0.3 test
> node --require source-map-support/register node_modules/@definitelytyped/dtslint/ types "@kesha-antonov/react-native-background-downloader"

dtslint@0.0.142
Error: /Users/philipsu/Code/DefinitelyTyped/types/kesha-antonov__react-native-background-downloader/kesha-antonov__react-native-background-downloader-tests.ts:5:8
ERROR: 5:8    expect  TypeScript@5.0 compile error: 
Cannot find module '@kesha-antonov/react-native-background-downloader' or its corresponding type declarations.
ERROR: 21:15  expect  TypeScript@5.0 compile error: 
Binding element 'expectedBytes' implicitly has an 'any' type.
ERROR: 22:15  expect  TypeScript@5.0 compile error: 
Parameter 'percent' implicitly has an 'any' type.
ERROR: 24:12  expect  TypeScript@5.0 compile error: 
Parameter 'error' implicitly has an 'any' type.

    at testTypesVersion (/Users/philipsu/Code/DefinitelyTyped/node_modules/@definitelytyped/dtslint/src/index.ts:235:11)
    at runTests (/Users/philipsu/Code/DefinitelyTyped/node_modules/@definitelytyped/dtslint/src/index.ts:197:7)
    at main (/Users/philipsu/Code/DefinitelyTyped/node_modules/@definitelytyped/dtslint/src/index.ts:83:5)
```
The test file looks like this:
```Typescript
import RNBackgroundDownloader, {
    checkForExistingDownloads,
    DownloadTask,
    ensureDownloadsAreRunning,
} from '@kesha-antonov/react-native-background-downloader';

// Set global headers for downloader
RNBackgroundDownloader.setHeaders({
    'x-custom': 'Custom Header',
    'x-custom-2': 'Another Custom Header',
});

const task = RNBackgroundDownloader.download({
    id: 'file123',
    url: 'https://link-to-very.large/file.zip',
    destination: `${RNBackgroundDownloader.directories.documents}/file.zip`,
    headers: {
        'x-custom-3': 'a-third-header',
    },
})
    .begin(({ expectedBytes }) => {})
    .progress(percent => {})
    .done(() => {})
    .error(error => {});

const taskFuncTest = (task: DownloadTask) => {
    // Check task state
    switch (task.state) {
        case 'DONE':
        case 'DOWNLOADING':
        case 'PAUSED':
        case 'FAILED':
        case 'STOPPED':
            break;
    }

    // Pause the task
    task.pause();

    // Resume after pause
    task.resume();

    // Cancel the task
    task.stop();
};

checkForExistingDownloads();
ensureDownloadsAreRunning();

taskFuncTest(task);
```
What's the right way to get the test to actually find the `index.d.ts` that I'm proposing for this package?</div>

--- a/README.md
+++ b/README.md
@@ -614,8 +614,10 @@ When `dts-gen` is used to scaffold a scoped package, the `paths` property has to
 
 ```json
 {
-  "paths": {
-    "@foo/*": ["foo__*"]
+  "compilerOptions": {
+    "paths": {
+      "@foo/*": ["foo__*"]
+    }
   }
 }
 ```

